### PR TITLE
[WIP] Solr 5.5.1

### PR DESCRIPTION
**Pull request by fosskers**
Solr 5.5.1 is the most recent version of the 5.x branch. For reasons outside my control, I'm unable to use Solr 5.0 or 6, so I've provided fixes here to update `solrs` to use Solr 5.5 instead.

#### Testing
I'm testing with a Solr 5.5 docker image. It can be set up with:
```console
docker pull solr:5.5
docker run --name solrs_test -d -p 8983:8983 -t solr:5.5
docker exec -it --user=solr solrs_test bin/solr create_core -c collection1
```
`sbt test` will succeed at compiling, and past most tests. I don't unfortunately have a SolrCloud/Zookeeper setup, so I expected those tests to fail. Here are the results on my machine:
```console
[info] Run completed in 23 seconds, 307 milliseconds.
[info] Total number of tests run: 52
[info] Suites: completed 10, aborted 3
[info] Tests: succeeded 47, failed 5, canceled 0, ignored 0, pending 0
[info] *** 3 SUITES ABORTED ***
[info] *** 5 TESTS FAILED ***
[error] Failed tests:
[error] 	io.ino.solrs.PingStatusObserverIntegrationSpec
[error] 	io.ino.solrs.CloudSolrServersUninitializedIntegrationSpec
[error] Error during tests:
[error] 	io.ino.solrs.CloudSolrServersIntegrationSpec
[error] 	io.ino.solrs.AsyncSolrClientIntegrationSpec
[error] 	io.ino.solrs.AsyncSolrClientCloudIntegrationSpec
[error] (test:test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 24 s, completed Jun 24, 2016 12:57:23 PM
```
Are these failures "okay" given that I don't have a cloud setup?

### CODE REVIEW

File: src/main/scala/io/ino/solrs/AsyncSolrClient.scala [hunk 1]
```diff
@@ -243,7 +243,7 @@ class AsyncSolrClient private (val loadBalancer: LoadBalancer,
     val promise = scala.concurrent.promise[QueryResponse]
     val startTime = System.currentTimeMillis()
 
-    val url = solrServer.baseUrl + getPath(q) + ClientUtils.toQueryString(wparams, false)
+    val url = solrServer.baseUrl + getPath(q) + wparams.toQueryString
```
Comment (magro) [inline src/main/scala/io/ino/solrs/AsyncSolrClient.scala]: I'd prefer to keep support for solr 5.0, therefore we should use ClientUtils.toQueryString if available and otherwise wparams.toQueryString (both via reflection).

File: src/main/scala/io/ino/solrs/SolrServers.scala [hunk 2]
```diff
@@ -274,16 +274,11 @@ object CloudSolrServers {
     }
   }
 
-  private def serverStatus(replica: Replica): ServerStatus = replica.get(ZkStateReader.STATE_PROP) match {
-    case ZkStateReader.ACTIVE => Enabled
-    case ZkStateReader.RECOVERING => Disabled
-    case ZkStateReader.RECOVERY_FAILED => Failed
-    case ZkStateReader.DOWN => Failed
-    case default =>
```
Comment (magro) [inline src/main/scala/io/ino/solrs/SolrServers.scala]: Why did you remove the default case?

**Comment by magro** [comment]
Thanks for the PR! The tests are self contained, therfore a simple sbt test should work, no docker needed. The failed tests show real issues that need to be resolved.

File: src/main/scala/io/ino/solrs/SolrServers.scala (hunk 2 above)
Comment (fosskers) [inline src/main/scala/io/ino/solrs/SolrServers.scala]: It's no longer needed, as those are the only four states possible.

**Comment by fosskers** [comment]
Really? No outside solr instance is necessary?

File: src/main/scala/io/ino/solrs/SolrServers.scala (hunk 2 above)
Comment (magro) [inline src/main/scala/io/ino/solrs/SolrServers.scala]: Ok, fine.

**Comment by magro** [comment]
Really. You could also check .travis.yml if it instructs travis to do anything special. The current master should work with sbt test...

**Comment by fosskers** [comment]
True, an `sbt test` on the current master does work.

**Comment by fosskers** [comment]
I've talked to my team about this. We're going to stick with raw SolrJ for the team being, but may transition to `solrs` depending on what our clients are looking for. There's also the possibility of forking this internally, to avoid supporting 5.0.

**Comment by magro** [comment]
JFYI, I just released solrs 1.5.0 with support for solr 6.1.0,works with 5.5.x as well.

**Comment by fosskers** [comment]
Thanks, we'll take a look.
